arroyo-storage/src/lib.rs

```diff
@@ -119,6 +119,28 @@ fn matchers() -> &'static HashMap<Backend, Vec<Regex>> {
     })
 }
 
+macro_rules! retry {
+    ($e:expr) => {{
+        use std::thread::sleep;
+        use std::time::Duration;
+        let mut retries = 0;
+        let max_retries: u32 = 5;
+        let backoff_factor: u64 = 2;
+        loop {
+            match $e {
+                Ok(value) => break Ok(value),
+                Err(e) if retries < max_retries => {
+                    retries += 1;
+                    println!("Error: {}. Retrying...", e);
+                    let backoff_time = Duration::from_millis(10 * backoff_factor.pow(retries));
```

Comment:
What do you think about placing a bound on this so it can't get way too large (for example if we later increase the number of retries)?